src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1688,7 +1688,18 @@ void TorrentHandle::handleTrackerErrorAlert(const lt::tracker_error_alert *p)
 
     m_trackerInfos[trackerUrl].lastMessage = message;
 
-    m_session->handleTorrentTrackerError(this, trackerUrl);
+    // Starting with libtorrent 1.2.x each tracker has multiple local endpoints from which
+    // an announce is attempted. Some endpoints might succeed while others might fail.
+    // Emit the signal only if all endpoints have failed. TrackerEntry::isWorking() returns
```

Comment:
The comment of how `TrackerEntry::isWorking()` works should be documented at trackerentry.cpp, no?

Reply:
I don't know. I did it here to help any reader on why I seem to check only once but I say "multiple endpoints" (in the comment).